modules/openapi-generator/src/main/resources/typescript-angular/param.mustache

```diff
@@ -0,0 +1,56 @@
+/**
+ * Standard parameter styles defined by OpenAPI spec
+ */
+export type ParamStyle =
+  | 'matrix'
+  | 'label'
+  | 'form'
+  | 'simple'
+  | 'spaceDelimited'
+  | 'pipeDelimited'
+  | 'deepObject'
+  ;
+
+/**
+ * Convenience: simulate ParameterStyle enum
+ */
+export const ParamStyle = {
```

Comment:
nice!

how about exporting the type as a string enum

```typescript
enum ParamStyle {
  matrix,
  ...
}
```
?

see the discussion in https://github.com/OpenAPITools/openapi-generator/issues/3451

Reply:
same below

Reply:
Hmm... don't forget about the associated union-type:
```typescript
export type PathParameterStyle =
  | 'matrix'
  | 'label'
...
```

This allows me to write easy to read code:
```typescript
const style: ParamStyle = 'matrix';
```

Using enums, users'd have to write the loooooong (and thus annoying):
```typescript
const style: ParamStyle = ParamStyle.matrix;
```


The const object is solely there if anyone wants to iterate over all possible `ParamStyle`s and could be dropped without loss.
Please also note: it's exported `as const` so it's effectively the same as the enum.
And finally: enums are nowhere to be found in @angular/core. (union-types all the way, iirc).

If I'd have to choose, I'd rather keep the union and drop the const (without introducing the enum).

But: your decision, please advise.

Reply:
p.s.: afaik, you cannot have an enum + union type: this does not compile because enum types cannot be merged with other types, just other enums

Reply:
Since string enums should become the default for the genrated angular code, but currently its still not the default:

https://github.com/OpenAPITools/openapi-generator/blob/18bb5466bb3c32f873337a8bc9f2fe6687490cb4/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAngularClientCodegen.java#L80

I suggest to use string enums here.

> enums are nowhere to be found in @angular/core. (union-types all the way, iirc).

I wonder whether this has historical reasons (pre TS 2.4) or whether there is another compelling reason. If you find something, let me know, in the end either way would work for the code here.